subprojects/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ProgressEvents.groovy

```diff
@@ -282,32 +282,51 @@ class ProgressEvents implements ProgressListener {
         Operation child(String displayName) {
             def child = children.find { it.descriptor.displayName == displayName }
             if (child == null) {
-                throw new AssertionFailedError("No operation with display name '$displayName' found in children of '$descriptor.displayName': $children")
+                throw new AssertionFailedError("No operation with display name '$displayName' found in children of '$descriptor.displayName':\n${describeList(children)}")
             }
             return child
         }
 
         Operation descendant(String displayName) {
             def found = [] as List<Operation>
-            def descendantsText = ''
             def recurse
-            recurse = { List<Operation> children, int level = 0 ->
```

Comment:
Nope, if joined, then the reference to `recurse` inside the closure doesn't compile.